[CI] IOUtilsTests testSyncWithLinksThrowExceptionWhenDestinationIsNotWritable failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6986 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/e7jwyvvdz3epa)
- [elasticsearch-periodic-platform-support #6979 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/bbsjvfxdsc4ec)
- [elasticsearch-periodic-platform-support #6967 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/evql2vmchx4po)

**Reproduction Line:**
```
null
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.test.cluster.util.IOUtilsTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSyncWithLinksThrowExceptionWhenDestinationIsNotWritable),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected java.io.UncheckedIOException to be thrown, but nothing was thrown
```

**Issue Reasons:**
- [main] 3 consecutive failures in step windows-2022_checkpart1_platform-support-windows
- [main] 3 failures in test testSyncWithLinksThrowExceptionWhenDestinationIsNotWritable (2.7% fail rate in 113 executions)
- [main] 3 failures in step windows-2022_checkpart1_platform-support-windows (100.0% fail rate in 3 executions)
- [main] 3 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/test/test-clusters/src/test/java/org/elasticsearch/test/cluster/util/IOUtilsTests.java
+++ b/test/test-clusters/src/test/java/org/elasticsearch/test/cluster/util/IOUtilsTests.java
@@ -9,6 +9,7 @@
 
 package org.elasticsearch.test.cluster.util;
 
+import org.junit.Assume;
 import org.junit.Test;
 
 import java.io.IOException;
@@ -56,6 +57,8 @@ public class IOUtilsTests {
 
     @Test
     public void testSyncWithLinksThrowExceptionWhenDestinationIsNotWritable() throws IOException {
+        Assume.assumeFalse("On Windows read-only directories are not supported", OS.current() == OS.WINDOWS);
+
         // given
         Path sourceDir = Files.createTempDirectory("sourceDir");
         Files.createFile(sourceDir.resolve("file1.txt"));

